app/Http/Controllers/API/v1/TransportController.php

```diff
@@ -402,25 +403,43 @@ public function create(Request $request): JsonResponse {
                                             'destination' => ['required', 'numeric'],
                                             'departure'   => ['required', 'date'],
                                             'arrival'     => ['required', 'date'],
-                                            'force'       => ['nullable', 'boolean']
+                                            'force'       => ['nullable', 'boolean'],
+                                            'with'        => ['nullable', 'array'],
                                         ]);
+        if (isset($validated['with'])) {
+            if (count($validated['with']) > 10) {
+                return $this->sendError('You can only checkin for up to 10 users at once.', 400);
+            }
+            $withUsers = User::whereIn('id', $validated['with'])->get();
+            foreach ($withUsers as $user) {
+                if (!Auth::user()?->can('checkin', $user)) {
+                    return $this->sendError('You are not allowed to checkin for the given user.', 401);
```

Comment:
This would be 403. 401 is unauthorized (not logged in)

Reply:
Ideas for more usability:

1. We should collect all "illegal" users from this request and return an array of disallowed user ids, so that the user can adapt the request
2. Maybe we should not even cancel the checkin request but rather return an information about the failed request afterwards.

Option 2 is very much up for debate because the frontend should already filter which users are allowed to be checked in, etc.